app/services/database_manager.py

```diff
@@ -0,0 +1,231 @@
+    """
+    client.recreate_collection(
+        collection_name=collection_name,
+        vectors_config=rest.VectorParams(size=384, distance=rest.Distance.COSINE)
+    )
+
+
+def store_embeddings(collection_name, text_chunks, embeddings):
+    """
+    Stores embeddings and their metadata in the Qdrant collection.
+
+    Parameters:
+    collection_name (str): The name of the collection.
+    text_chunks (list): A list of text chunks.
+    embeddings (list): A list of embeddings.
+
+    Returns:
+    None
+    """
+    points = [
+        rest.PointStruct(id=i, vector=embedding.tolist(), payload={"text": text_chunk})
+        for i, (embedding, text_chunk) in enumerate(zip(embeddings, text_chunks))
+    ]
+    client.upsert(collection_name=collection_name, points=points)
+
+
+def process_and_store_text(pdf_path, collection_name):
+    """
+    Processes text from a PDF and stores embeddings in Qdrant.
+
+    Parameters:
+    pdf_path (str): The path to the PDF file.
+    collection_name (str): The name of the collection.
+
+    Returns:
+    None
+    """
+    # Extract text from the PDF
+    document = fitz.open(pdf_path)
+    full_text = ""
+    for page_num in range(len(document)):
+        page = document[page_num]
+        full_text += page.get_text()
+
+    # Partition text and generate embeddings
+    text_chunks = partition_text(full_text)
+    embeddings = generate_embeddings(text_chunks)
+
+    # Create collection and store embeddings
+    create_collection(collection_name)
+    store_embeddings(collection_name, text_chunks, embeddings)
+
+
+#==============================================================
+# Function to recover the most relevant text fragments
+
+def query_embeddings(query, collection_name='contract_text_embeddings', top_k=5):
+    """
+    Queries the Qdrant database to find the most relevant text fragments for a given query.
+
+    Parameters:
+    query (str): The query to find relevant text fragments for.
+    collection_name (str): The name of the collection to query.
+    top_k (int): The number of top results to retrieve.
+
+    Returns:
+    list: A list of relevant text fragments.
+    """
+    query_embedding = model.encode([query])
+    search_result = client.search(
+        collection_name=collection_name,
+        query_vector=query_embedding[0].tolist(),
+        top=top_k
+    )
+
+    relevant_texts = [hit.payload['text'] for hit in search_result]
+    return relevant_texts
+
+
+def search_similar_texts(collection_name, query_text, top_k=5):
+    """
+    Searches for the top K most similar texts in the specified Qdrant collection.
+
+    Parameters:
+    collection_name (str): The name of the collection.
+    query_text (str): The query text to search for.
+    top_k (int): The number of top similar texts to retrieve.
+
+    Returns:
+    list: A list of the top K most similar texts and their scores.
+    """
+    query_vector = model.encode([query_text])[0]
+    search_result = client.search(
```

Comment:
As a comment: check the different search options qdrant provides: cosine similarity,  euclidean distance , mmr, .etc...